### ISSUE TITLE
JS Windows should be tab modal, not application modal

### ISSUE THREAD

**Issue by garvankeeley** [brave/browser-laptop#1990]
**Describe the issue you encountered:**
Malicious sites will popup a JS modal dialog that blocks input, the browser has to be killed to close the window. You should be able to close the tab.

**Expected behavior:**
JS popup windows should be Tab-modal, not app-modal. Tab should be closable without interacting with the popup.

Mackeeper popups are an example of this.

**Comment by cezaraugusto** [comment]
@garvankeeley is anyone already working on it? /cc @diracdeltas

**Comment by diracdeltas** [comment]
cc @bridiver

**Comment by bridiver** [comment]
it's doable, be we'd have to rewrite electron's dialog handler to use an NSWindow instead of an NSAlert. afaik there is no way to make a non-window-modal NSAlert

**Comment by bridiver** [comment]
actually I take that back. The dialogs are initiated by ipc calls so there is no reason why we can't override and handle ourselves

**Comment by bridiver** [comment]
should also have a checkbox to prevent future dialog boxes for a site

**Comment by luixxiul** [comment]
Closing per https://github.com/brave/browser-laptop/pull/7107

**Comment by bsclifton** [comment]
@luixxiul this has been closed for some time, right? with https://github.com/brave/browser-laptop/pull/7107

**Comment by luixxiul** [comment]
I thought this hasn't, but you're right. thanks.
